homeassistant/components/todoist/calendar.py

```diff
@@ -622,10 +629,7 @@ async def async_get_events(
         tasks = self._coordinator.data
         if self._id is None:
             project_task_data = [
-                task
-                for task in tasks
-                if not self._project_id_whitelist
-                or task.project_id in self._project_id_whitelist
+                task for task in tasks if self.create_todoist_task(task) is not None
```

Comment:
Filtering by the project_id_whitelist was moved into `create_todoist_task` where the other 2 filters are applied.